### PR TITLE
personality: Add support for setting execution domain.

### PR DESCRIPTION
**Pull request by flouthoc**
Hi Team,

Following PR allows users to set execution domain via Linux personality. Please find a detailed discussion of 
**use-case:** https://github.com/opencontainers/runc/pull/3126#issuecomment-892161740
**upstream-PR:** https://github.com/opencontainers/runc/pull/3126 

References:
https://man7.org/linux/man-pages/man2/personality.2.html

### CODE REVIEW

File: cmd/podman/common/create.go [hunk 1]
```diff
@@ -498,6 +498,14 @@ func DefineCreateFlags(cmd *cobra.Command, cf *ContainerCLIOpts) {
 	)
 	_ = cmd.RegisterFlagCompletionFunc(variantFlagName, completion.AutocompleteNone)
 
+	personalityFlagName := "personality"
+	createFlags.StringVar(
+		&cf.Personality,
+		personalityFlagName, "",
+		"Configure execution domain using personality. (E.g LINUX/LINUX32)",
```
Comment (vrothberg) [inline cmd/podman/common/create.go:505]:
```suggestion
		"Configure execution domain using personality (e.g., LINUX/LINUX32)",
```

File: docs/source/markdown/podman-create.1.md [hunk 2]
```diff
@@ -691,6 +691,11 @@ Tune the host's OOM preferences for containers (accepts -1000 to 1000)
 #### **--os**=*OS*
 Override the OS, defaults to hosts, of the image to be pulled. For example, `windows`.
 
+#### **--personality**=*persona*
+
+Personality sets the execution domain via linux personality.
```
Comment (vrothberg) [inline docs/source/markdown/podman-create.1.md:696]: Please also point to a man page where users can read more on personalities.

**Review by vrothberg — COMMENTED** [review]
Thanks, the changes LGTM but a test would be nice. Also please make sure to describe the changes in the commit message.

**Comment by flouthoc** [comment]
@vrothberg Thanks :) resolved all feedback points. Waiting for tests to pass.

**Comment by rhatdan** [comment]
LGTM
Does crun and runc support personality at this time?
@giuseppe @mheon @vrothberg PTAL

**Comment by flouthoc** [comment]
@rhatdan `crun` has a support for this added by @giuseppe in https://github.com/containers/crun/pull/217 for `runc` there is a upstream pr under-review and discussions opencontainers/runc#3126

File: pkg/specgen/generate/oci.go [hunk 3]
```diff
@@ -285,6 +285,11 @@ func SpecGenToOCI(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Runt
 		}
 		g.AddMount(cgroupMnt)
 	}
+
+	if s.Personality != nil {
+		g.Config.Linux.Personality = s.Personality
```
Comment (giuseppe) [inline pkg/specgen/generate/oci.go:290]: this can just be `g.Config.Linux.Personality = s.Personality`

File: test/e2e/run_test.go [hunk 4]
```diff
@@ -1790,4 +1790,11 @@ WORKDIR /madethis`, BB)
 		_, err = strconv.Atoi(containerPID) // Make sure it's a proper integer
 		Expect(err).To(BeNil())
 	})
+
+	It("podman run check personality support", func() {
+		session := podmanTest.Podman([]string{"run", "--personality=LINUX32", "--name=testpersonality", ALPINE, "uname", "-a"})
```
Comment (giuseppe) [inline test/e2e/run_test.go:1795]: do we need to skip this test at the moment when runc is used?

Comment (flouthoc) [reply test/e2e/run_test.go:1795]: @giuseppe oh yes thanks for spotting this. Adding a skip.

File: docs/source/markdown/podman-create.1.md (hunk 2 above)
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-create.1.md:711]:
```suggestion
Personality sets the execution domain via Linux personality.
```

File: docs/source/markdown/podman-create.1.md [hunk 5]
```diff
@@ -706,6 +706,12 @@ Tune the host's OOM preferences for containers (accepts -1000 to 1000)
...
+Reference: https://man7.org/linux/man-pages/man2/personality.2.html
+Example: **--personality=LINUX32**.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:713]: Although small, to be consistent, this should be in the Examples section at the bottom of the man page.

File: docs/source/markdown/podman-run.1.md [hunk 6]
```diff
@@ -726,6 +726,12 @@ Tune the host's OOM preferences for containers (accepts values from **-1000** to
 #### **--os**=*OS*
 Override the OS, defaults to hosts, of the image to be pulled. For example, `windows`.
 
+#### **--personality**=*persona*
+
+Personality sets the execution domain via linux personality.
+Reference: https://man7.org/linux/man-pages/man2/personality.2.html
+Example: **--personality=LINUX32**.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:733]: See prior comments

**Comment by TomSweeneyRedHat** [comment]
Couple of man page changes, otherwise LGTM

**Comment by flouthoc** [comment]
@giuseppe @TomSweeneyRedHat @vrothberg resolved all them could you guys please take a look again.

File: docs/source/markdown/podman-create.1.md [hunk 7]
```diff
@@ -706,6 +706,11 @@ Tune the host's OOM preferences for containers (accepts -1000 to 1000)
 #### **--os**=*OS*
 Override the OS, defaults to hosts, of the image to be pulled. For example, `windows`.
 
+#### **--personality**=*persona*
+
+Personality sets the execution domain via linux personality.
+Reference: https://man7.org/linux/man-pages/man2/personality.2.html
```
Comment (vrothberg) [inline docs/source/markdown/podman-create.1.md:712]: It's more idiomatic to write `personality(2)` in man pages and to add that to the references at the very bottom of a manpage.

I suggest to remove this line and turn the one above into:
> Personality sets the execution domain via Linux personality(2).

File: docs/source/markdown/podman-run.1.md [hunk 8]
```diff
@@ -726,6 +726,11 @@ Tune the host's OOM preferences for containers (accepts values from **-1000** to
 #### **--os**=*OS*
 Override the OS, defaults to hosts, of the image to be pulled. For example, `windows`.
 
+#### **--personality**=*persona*
+
+Personality sets the execution domain via linux personality.
+Reference: https://man7.org/linux/man-pages/man2/personality.2.html
```
Comment (vrothberg) [inline docs/source/markdown/podman-run.1.md:732]: Same here.

File: docs/source/markdown/podman-run.1.md [hunk 9]
```diff
@@ -1776,6 +1781,12 @@ $ podman run --name container3 --requires container1,container2 -t -i fedora bas
 $ podman run -v /var/lib/design:/var/lib/design --group-add keep-groups ubi8
 ```
 
+### Configure execution domain for containers using personality flag.
```
Comment (vrothberg) [inline docs/source/markdown/podman-run.1.md:1784]:
```suggestion
### Configure execution domain for containers using personality flag
```

File: docs/source/markdown/podman-create.1.md [hunk 10]
```diff
@@ -1429,6 +1434,12 @@ $ podman start --attach container3
 $ podman create -v /var/lib/design:/var/lib/design --group-add keep-groups ubi8
 ```
 
+### Configure execution domain for containers using personality flag.
```
Comment (vrothberg) [inline docs/source/markdown/podman-create.1.md:1437]:
```suggestion
### Configure execution domain for containers using personality flag
```

File: test/e2e/run_test.go [hunk 11]
```diff
@@ -1790,4 +1790,15 @@ WORKDIR /madethis`, BB)
 		_, err = strconv.Atoi(containerPID) // Make sure it's a proper integer
 		Expect(err).To(BeNil())
 	})
+
+	It("podman run check personality support", func() {
+		// TODO: Remove this as soon as this is merged https://github.com/opencontainers/runc/pull/3126.
```
Comment (vrothberg) [inline test/e2e/run_test.go:1795]:
```suggestion
		// TODO: Remove this as soon as this is merged and made available in our CI https://github.com/opencontainers/runc/pull/3126.
```

**Review by vrothberg — COMMENTED** [review]
Thanks, @flouthoc!  Just mini nits, but LGTM.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Pull request merged**